lib/rules/require-expect.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @fileoverview Require the use of `expect` when using `assert` inside of a
+ * block or when passing `assert` to a function.
+ * @author Mitch Lloyd
+ */
+"use strict";
+
+var utils = require("../utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function (context) {
+    var EXCEPT_SIMPLE_ERROR_MESSAGE = "Should use `assert.expect` when using assertions outside of the top-level test callback",
+        ALWAYS_ERROR_MESSAGE = "Test is missing `assert.expect() call`",
```

Comment:
This should probably say `"Test is missing \`assert.expect()\` call".

Also, unfortunately this doesn't allow for the assert variable being renamed:

```js
test("Some test", function(mySpecialAssertVar) {});
```

You can/should use `utils.getAssertContextNameForTest()` on the CallExpression representing the beginning of a test, and then store that information so you can use it in the error message.